### DISCUSSION TITLE
Adapting Open Web UI Pipelines for RAG with PandasAI: Seeking Examples and Guidance

### DISCUSSION THREAD

**Discussion by daltonbc96** [open-webui/pipelines#183] [General]
Hello everyone,

I've been exploring the capabilities of Open Web UI and, after reviewing the official documentation, I couldn't find explicit support for creating retrieval-augmented generation (RAG) for datasets or databases using tools like PandasAI.

My goal is to utilize pipeline functionalities to handle user-uploaded documents, specifically CSV files, transform them into dataframes, and perform RAG on databases using pipelines. While I understand this may not be directly supported, I'm curious if anyone has successfully adapted Open Web UI for such purposes.

Has anyone here managed to implement RAG with datasets or databases using PandasAI in Open Web UI? If so, could you share example Python code or guidance on how to capture CSV files, convert them into dataframes, and conduct RAG on databases in the context of the pipeline?

Any help or pointers to relevant examples would be greatly appreciated!

**Comment by MrDick-11** [comment]
good question! I have seen this https://openwebui.com/t/ajayrajc/chat_with_csv
But it's not working
Standing behind you and wait for a solution)
